Dm 2726 - Disable ability to comment for public users.

bradjohnson92008 opened the pull request:
### JIRA issue link
https://agile6.atlassian.net/browse/DM-2726

## Description - what does this code do?
This code identifies if the current user is a "public" user and if so disables the ability to comment or reply to the Practice comment's section on the Practice show page.  User can still view previous comments but comments are read only and they can not create new comments on the practice.  Edit and delete links are disabled.

## Testing done - how did you test it/steps on how can another person can test it 
1.  Login to DM as a a non "ntlm" user.  i.e., non-gfe or cag user.  Following method should return nil for user in the application controller.  This will set the current user to a "public" user.
![image](https://user-images.githubusercontent.com/60527788/126236525-66a6213c-afe2-4625-8f1e-c1de5f78fdea.png)
2.  Navigate to a Practice show page that contains comments.  
3. Scroll down to the pages "Contact / Comment" section.  Since you are logged in as a public user you should see the INformation banner stating that "Comments and replies are disabled for retired practices and public users."
![image](https://user-images.githubusercontent.com/60527788/126236846-24bfd768-97a7-4380-b5b9-773aa3526a99.png)
4.  Scroll through the comments.  User should be able to view all comments, and expand or collapse "Replies".
5. Click on the Edit and Delete icons.  These icons should be disabled - i.e., no action occurs when clicking them.
6.  Log off and log back on as an NTLM (VHA network) user.  
7. Navigate to a practice that is not retired.
8. Comment and reply functionality should enabled and all functionality wihin commontator should work correctly.


## Acceptance criteria
Public users don't see comment textbox or role radio button
Comments are read only
Test on Dev when public
Test on STG when public
Release to Prod

## Definition of done
- [ ] Unit tests written (if applicable)
- [ ] e2e/accessibility tests written (if applicable)
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating JIRA issue
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

emyl3 reviewed (changes requested):
The styles are looking a little off for me. Is anyone else seeing this too?
<img width="1247" alt="Screen Shot 2021-07-20 at 4 01 48 PM" src="https://user-images.githubusercontent.com/20211771/126397474-2502da60-ac88-4e8e-b138-7617941d9add.png">


Also, I am seeing the edit button when I'm a different user for a comment that I didn't write and this user is not an admin.
<img width="1090" alt="Screen Shot 2021-07-20 at 4 20 53 PM" src="https://user-images.githubusercontent.com/20211771/126397575-65e0c916-198a-41a9-8a9c-f0e09e2f9600.png">
Should we just hide the "Reply" button and thumbs up button if they are a public user?

bradjohnson92008 commented on app/views/commontator/comments/_form.html.erb:
```diff
@@ -33,70 +33,72 @@
-    </div>
-    <div class="text-left margin-top-2">
-      <% if config.new_comment_style == :t && new_record && comment.parent.nil? %>
+    <% is_public_user = session[:user_type] === 'public' ? true : false %>
```
true.. :)

bradjohnson92008 commented on db/schema.rb:
```diff
@@ -10,8 +10,7 @@
 
-ActiveRecord::Schema.define(version: 2021_06_02_143652) do
-
+ActiveRecord::Schema.define(version: 2021_06_10_160757) do
```
Got it.

JoshingYou1 commented:
> The styles are looking a little off for me. Is anyone else seeing this too?
Yeah, the styling is off on my local as well
![Screen Shot 2021-07-21 at 2 00 14 PM](https://user-images.githubusercontent.com/34111449/126537539-d4ac8999-0715-4a4e-a25b-82ae7d2e6304.png)

> Should we just hide the "Reply" button and thumbs up button if they are a public user?
I think that's a good idea. We should hide anything the user cannot actually interact with

JoshingYou1 commented on db/schema.rb:
```diff
@@ -217,6 +216,10 @@
+  create_table "clinical_id", id: false, force: :cascade do |t|
+    t.bigint "id"
+  end
+
```
Are we using this?

bradjohnson92008 replied:
No - I removed that create_table code

JoshingYou1 reviewed (changes requested):
Aside from some of the styling, just left one comment Brad. Nice work!

bradjohnson92008 commented:
Added more code to remove all the icons/buttons ..etc that the user can not interact with.  Also removed link for comment creator - you can still see the comment creators name and job title.. but removed the link to the profile page.

emyl3 reviewed (changes requested):
Found a couple of lines of code we can remove.

Thoughts on left aligning the "Show X reply" button. It looks off not being left aligned without the other buttons.
<img width="704" alt="Screen Shot 2021-07-22 at 10 17 18 AM" src="https://user-images.githubusercontent.com/20211771/126664237-bb6eacb3-d76d-46a8-aaca-911cf99397a7.png">

bradjohnson92008 commented on app/views/commontator/comments/_show.html.erb:
```diff
@@ -93,33 +95,34 @@
                              class: is_deleted ? "fas fa-trash-restore-alt fa-lg margin-left-1" : "fas fa-trash-alt fa-lg delete-comment-icon margin-left-1",
                              remote: true %>
             <% else %>
```
Got it.

bradjohnson92008 commented on app/views/commontator/comments/_show.html.erb:
```diff
@@ -147,9 +150,9 @@
               %>
             </span>
             <% else %>
```
Got it.

bradjohnson92008 commented:
I left aligned the Show/Hide Replies btn/link since we are not displaying the other icons.

emyl3 reviewed (approved):
Looks good to me! Great work, Brad!

JoshingYou1 reviewed (approved):
Works as intended. Great job Brad!!
